src/python/turicreate/toolkits/_tree_model_mixin.py

```diff
@@ -317,6 +317,7 @@ def _get_summary_struct(self):
 
     def _export_coreml_impl(self, filename, context):
         tc_version_info = _coreml_utils._get_tc_version_info()
+        tc_version_info['type'] = context['class']
         if 'user_defined' not in context:
```

Comment:
How are we testing this?

Reply:
I'm thinking of how to do that, specifically how to obtain the class name for asserting the `type` in https://github.com/apple/turicreate/blob/46d5a4de4388dd72d445dfe69fa6064a8e5cba1d/src/python/turicreate/test/test_coreml_export.py#L71 since `_test_coreml_export()` is run for all classifiers and regressors.

Reply:
It works on manual testing obviously but we'd still want to update the testing suite.

Reply:
Added `type` field check in test.

Reply:
It looks like `_coreml_utils.py` already provides a function that adds the toolkit type to this dictionary:
https://github.com/apple/turicreate/blob/b12f67fb523b26cbbff263f6260d0cf2674d8647/src/python/turicreate/toolkits/_coreml_utils.py#L27
Could we be using that instead of the direct calls to `_get_tc_version_info`?
This is in fact what the style transfer toolkit (the motivating example from the bug) uses:

Reply:
The reason I didn't use `_get_model_metadata` is because we don't have `metadata` given in `_export_coreml_impl`. 

I'm wondering if there is a way to get `metadata` from `self.__proxy__` in python as `_xgboost_export_as_model_asset(...)` does in:

https://github.com/apple/turicreate/blob/9815427e8dba58e3ead0807552238644a73b68c4/src/toolkits/coreml_export/coreml_extension.cpp#L33

Reply:
It looks like it's okay for `metadata` to be `None`

Reply:
Oh that's right! Would it be ok if I also changed:

`def _get_model_metadata(model_class, metadata, version=None):
`

to

`def _get_model_metadata(model_class, metadata=None, version=None):
`

for better readability?

Reply:
Seems reasonable to me!